### PR TITLE
Refine sunrise and sunset #or-environment

### PR DESCRIPTION
**Pull request by twpol**
Roadmap: https://trello.com/c/6t0KwueQ/388-refine-sunrise-and-sunset-times
Discussion: http://www.elvastower.com/forums/index.php?/topic/34984-build-a-better-skydome-terragen-2/

**Changes**

- Load sun rise/set times from MSTS environment files 654dfd15964f60f967e9c5ed38158c2986e6ff55
- Apply them to the sun position calculation 5c518176eac2cb920f468d8c4b2924cb9142151b
- Fix the movement of the sun in the sky 719c7376d40240a807db273b7c6564b2c85e3c48

Includes fix for https://bugs.launchpad.net/or/+bug/2003001

### CODE REVIEW

**Review by pzgulyas — DISMISSED** [review]
I see no problems with this.

File: Source/RunActivity/Viewer3D/Common/SunMoonPos.cs [hunk 1]
```diff
@@ -36,153 +34,171 @@ static class SunMoonPos
...
+            var lunarAltSin = (Math.Sin(latitude) * Math.Sin(declination)) + (Math.Cos(latitude) * Math.Cos(lunarHourAngle));
+            var lunarAltitude = Math.Asin(lunarAltSin);
+
             // Lunar Azimuth, radians from its Cosine
-            double lunarAltCos = Math.Cos(lunarAltitude);
-            double hourAngleSin = Math.Sin(lunarHourAngle);
-            double lunarAzCos = (Math.Sin(Declination) - lunarAltSin * Math.Sin(latitude)) / (lunarAltCos * Math.Cos(latitude));
+            var lunarAltCos = Math.Cos(lunarAltitude);
+            var hourAngleSin = Math.Sin(lunarHourAngle);
+            var lunarAzCos = (Math.Sin(declination) - (lunarAltSin * Math.Sin(latitude))) / (lunarAltCos * Math.Cos(latitude));
```
[the remark this answers is not in the document]
Comment (twpol) [reply Source/RunActivity/Viewer3D/Common/SunMoonPos.cs:198]: These parentheses are to satisfy the default-enabled StyleCop rule `SA1407`: Arithmetic expressions should declare precedence

(I think I ran `dotnet format` on the file which will auto-fix such things)

File: Source/RunActivity/Viewer3D/Common/SunMoonPos.cs [hunk 2]
```diff
@@ -36,153 +34,171 @@ static class SunMoonPos
...
+
+            var solarZenithCosine = (Math.Sin(latitude) * Math.Sin(solarDeclination)) + (Math.Cos(latitude) * Math.Cos(solarDeclination) * Math.Cos(solarHourAngle));
+            var solarZenithSine = Math.Sin(Math.Acos(solarZenithCosine));
+            var solarAzimuthCosine = (Math.Sin(solarDeclination) - (solarZenithCosine * Math.Sin(latitude))) / (Math.Sin(Math.Acos(solarZenithCosine)) * Math.Cos(latitude));
+            var solarAzimuthSine = -Math.Sin(solarHourAngle) * Math.Cos(solarDeclination) / Math.Sin(Math.Acos(solarZenithCosine));
+
+            sunDirection.X = (float)solarZenithSine * (float)solarAzimuthSine;
             sunDirection.Y = (float)solarZenithCosine;
-            sunDirection.Z = -(float)solarAzimuthCosine;
+            sunDirection.Z = -(float)solarZenithSine * (float)solarAzimuthCosine;
```
[the remark this answers is not in the document]
Comment (twpol) [reply Source/RunActivity/Viewer3D/Common/SunMoonPos.cs:99]: Hmm, honestly not sure why the code ended up like this :)

I have changed it

**Review by pzgulyas — APPROVED** [review]
Looks fine to me.

**Pull request merged**